cscs-checks/tools/profiling_and_debugging/intel_advisor_roofline.py

```diff
@@ -0,0 +1,161 @@
+class IntelRooflineTest(rfm.RegressionTest):
+    '''This test checks the values reported by Intel Advisor's roofline model:
+    https://software.intel.com/en-us/intel-advisor-xe
+
+    The roofline model is based on GFLOPS and Arithmetic Intensity (AI):
+      "Self GFLOPS" = "Self GFLOP" / "Self Elapsed Time"
+      "Self GB/s" = "Self Memory GB" / "Self Elapsed Time"
+      "Self AI" = "Self GFLOPS" / "Self GB/s"
+
+    While a roofline analysis flag exists ('advixe-cl -collect roofline'), it
+    may not be used to collect data on MPI applications; in that case, the
+    survey and flops analysis must be collected separately: first run a survey
+    analysis ('advixe-cl -collect survey') and then run a tripcounts+flops
+    analysis ('advixe-cl -collect tripcounts -flop') using the same project
+    directory for both steps.
+    '''
+    def __init__(self, repeat, datalayout):
+        super().__init__()
+        self.name = 'Intel_Roofline_%s_%s' % (repeat, datalayout)
+        self.descr = 'repeat=%s' % repeat
+        self.valid_systems = ['daint:mc', 'dom:mc']
+        # Reporting MFLOPS is not available on Intel Haswell cpus, see
+        # https://www.intel.fr/content/dam/www/public/us/en/documents/manuals/
+        # 64-ia-32-architectures-software-developer-vol-1-manual.pdf
+        self.valid_prog_environs = ['PrgEnv-intel']
+        self.prgenv_flags = {
+            'PrgEnv-intel': ['-O2', '-g', '-std=c++11'],
+        }
+        self.sourcesdir = os.path.join(self.current_system.resourcesdir,
+                                       'roofline', 'intel_advisor')
+        self.build_system = 'Make'
+        self.prebuild_cmd = [
+            'sed -e "s-XXXX-%s-" -e "s-YYYY-%s-" %s &> %s' %
+            (repeat, datalayout, 'roofline_template.cpp', '_roofline.cpp')
+        ]
+        self.num_tasks = 1
+        self.num_tasks_per_node = 1
+        self.num_cpus_per_task = 1
+        self.variables = {
+            'OMP_NUM_THREADS': str(self.num_cpus_per_task),
+            'CRAYPE_LINK_TYPE': 'dynamic',
+        }
+        self.pre_run = [
+            # Testing with advisor/2018:
+            #   advisor/2019 is broken on dom ("Exceeded job memory limit"),
+            #   and advisor/2019 is not installed on daint,
+            'source $INTEL_PATH/../advisor_2018/advixe-vars.sh',
+            'advixe-cl -help collect |head -20',
+        ]
+        self.executable = 'advixe-cl'
+        self.exe = './roof.exe'
+        self.roofdir = './roof.dir'
+        self.executable_opts = [
+            '--collect survey --project-dir=%s --search-dir src:rp=. '
+            '--data-limit=0 --no-auto-finalize --trace-mpi -- %s ' %
+            (self.roofdir, self.exe)
+        ]
+        self.version_rpt = 'Intel_Advisor_version.rpt'
+        self.post_run = [
+            # collecting the performance data for the roofline model is a 2
+            # steps process:
+            'srun %s --collect tripcounts --flop --project-dir=%s '
```

Comment:
The most portable would be to define `post_run` in the `setup()` and instead of hardcoding `srun` to do the following:

```python
def setup(self, ...):
    super().setup(self, ...)
    launcher_cmd = self.job.launcher.command(self.job)
    self.post_run = ['%s blah blah' % launcher_cmd]
```

Reply:
👍